tests/testsuite/standard_lib.rs

```diff
@@ -250,31 +405,71 @@ fn check_std() {
         .file(
             "tests/t1.rs",
             r#"
-            #[test]
-            fn t1() {
-                assert_eq!(1, 2);
-            }
+                #[test]
+                fn t1() {
+                    assert_eq!(1, 2);
+                }
             "#,
         )
         .build();
 
-    cargo_build_std(&p, "check -v --all-targets", "std").run();
-    cargo_build_std(&p, "check -v --all-targets --profile=test", "std").run();
+    p.cargo("check -v --all-targets")
+        .build_std()
+        .target_host()
+        .run();
+    p.cargo("check -v --all-targets --profile=test")
+        .build_std()
+        .target_host()
+        .run();
 }
 
+#[cargo_test]
 fn doctest() {
+    if !setup() {
+        return;
+    }
     let p = project()
         .file(
             "src/lib.rs",
             r#"
-            /// Doc
-            /// ```
-            /// assert_eq!(1, 1);
-            /// ```
-            pub fn f() {}
+                /// Doc
+                /// ```
+                /// assert_eq!(1, 1);
+                /// ```
+                pub fn f() {}
             "#,
         )
         .build();
 
-    cargo_build_std(&p, "test --doc -v", "std").run();
+    p.cargo("test --doc -v").build_std().target_host().run();
 }
+
+// FIXME: set up a dedicated builder to run a full integration test?
```

Comment:
Do you want to leave this commented for now?  I'm fine with that, this test isn't super important right now.

But what would be a good long-term strategy?  Can we afford more builders on CI?  I don't want to be too greedy with our neighbors.  (Maybe we should prune some PR builders, and only run the full suite on merge?)  Also, building cargo itself multiple times in separate builders seems unnecessary.  I'd be fine if a job runs a little longer.  The thing that I think would be nice is to make it easier to run these "slow" tests separately (like the resolver tests were separated), particularly for local development.